src/Kunstmaan/AdminBundle/Helper/Security/OAuth/OAuthUserCreatorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Kunstmaan\AdminBundle\Helper\Security\OAuth;
+
+/**
+ * Interface OAuthUserCreatorInterface
+ */
+interface OAuthUserCreatorInterface
+{
+    /**
+     * Returns an implementation of AbstractUser defined by the $userClass parameter.
+     * Checks if there already exists an account for the given googleId or email. If yes updates
+     * the access levels accordingly and returns that user. If no creates a new user with the
+     * configured access levels.
+     *
+     * Returns Null if email is not in configured domains
+     *
+     * @param string email
+     * @param string googleId
+     *
+     * @return AbstractUser Implementation
```

Comment:
`@return mixed AbstractUser Implementation` — this way IDE won’t expect to return `AbstractUser` instance. Maybe it should be a `UserInterface` instance from FOS?